HelpSource/Classes/Collection.schelp

```diff
@@ -304,10 +304,94 @@ List[1, 2, 3, 3, 4, 3, 4, 3].occurrencesOf(3);
 ::
 
 method::sum
-Answer the sum of the results of strong::function:: evaluated for each item in the receiver. The function is passed two arguments, the item and an integer index.
+Answer the sum of the items in the collection, optionally applying a link::Classes/Function:: to each item before summing. When provided, the link::Classes/Function:: is passed two arguments: the item and an integer index.
+
+code::
+(0..8).sum { |item| 1 / (2 ** item) }
+// only the item is used; index is unused.
+::
+
+The code above is mathematically equivalent to:
+math::
+\sum_{i=0}^{n} \frac{1}{2^{x_i}} \quad \text{for} \quad \{x_0, x_1, \ldots, x_n\} \quad (n=8)
+::
+(emphasis::In the mathematic formula, the argument name:: teletype::item:: emphasis::in sclang code is represented as:: math::x_i::.)
+
+discussion::
+The sum of all items in a collection can be expressed mathematically as:
+
+table::
+##teletype::
+[x_0, x_1, ..., x_n].sum // pseudo-code
+::
+equivalent to:
+math::
+\text{Sum} \{ x_0, x_1, \ldots, x_n \} = \prod_{i=0}^{n} x_i
+::
+::
+
+Example:
+code::
+List[1, 2, 3, 4].sum
+::
+
+method::product
+Answer the product of the items in the collection, optionally applying a link::Classes/Function:: to each item before multiplication. When provided, the link::Classes/Function:: receives two arguments: the item and its integer index.
```

Comment:
I think this could also be structured like the `sum` documentation above.

Reply:
Partially done, except for the detailed step-by-step explanation. Let's see how the others react. If the detailed step-by-step explanation is preferred, I will create text for `.product` and `.mean` based on your text for `.sum`. See https://github.com/supercollider/supercollider/pull/6537#issuecomment-2690728941.